modules/react/common/lib/utils/changeFocus.ts

```diff
@@ -18,7 +18,8 @@ export const changeFocus = (element: unknown) => {
 
       element.dispatchEvent(event);
     }
-
-    element.focus();
   }
+
+  // Even if the element is not an HTMLElement, we should still attempt to focus it.
+  element.focus?.();
```

Comment:
What other element type are you getting? If not `HTMLElement`, is there additional instances we should test? Also, why not remove the `if` block entirely? You've fixed the focus change, but not focus styling.

Reply:
That's a good point! Why does this function care what `element` is an instanceof at all, so long as the methods are implemented? Updated.

The [react-select](https://www.npmjs.com/package/react-select) library returns a custom ref with an extended API for ["advanced" usages](https://react-select.com/advanced#accessing-internals), including the basics that you'd expect on a standard ref (like focus):
<img width="880" alt="image" src="https://user-images.githubusercontent.com/950986/236900156-8342ef19-5da2-4b32-b78b-fecf6ea48f88.png">

Reply:
The `if(element instanceof ...)` is for Typescript because the type is `unknown`. It doesn't really matter what the value is, but Typescript wants to know that what is handed to it has a callable property called 'focus'.

Reply:
The `if(element instanceof ...)` is for Typescript because the type is `unknown`. It doesn't really matter what the value is, but Typescript wants to know that what is handed to it has a callable property called 'focus'.

Reply:
You're change works in JavaScript, but it fails the TypeScript compiler because TypeScript still thinks `element` is `unknown` and that the methods are not callable.

We need to either cast the instances of `element` OR place a proper guard in the place of `instanceof` to grant both runtime and compiletime safety.

Reply:
I've added a guard to please TypeScript and avoid any runtime errors.